keras/layers/core/permute.py

```diff
@@ -32,15 +32,14 @@ class Permute(Layer):
 
   ```python
   model = Sequential()
-  model.add(Permute((2, 1), input_shape=(10, 64)))
-  # now: model.output_shape == (None, 64, 10)
-  # note: `None` is the batch dimension
+  model.add(Permute((1, 3, 2), input_shape=(30, 10, 64)))
+  # now: model.output_shape == (None, 30, 64, 10)
+  # note: `30` is the batch dimension
   ```
 
   Args:
-    dims: Tuple of integers. Permutation pattern does not include the
-      samples dimension. Indexing starts at 1.
-      For instance, `(2, 1)` permutes the first and second dimensions
+    dims: Tuple of integers. Expects a permutation of dimensions. len(ndims) should be equal to input_shape dimensions.
```

Comment:
Please wrap the long line here.

Reply:
Also, it not just that they should have the same length, they should have same elements, but just in different order.

Reply:
Yaa, soo i wanted to include here that dims will have the same elements as the input_shape parameter and each element in dims specifies the index in input_shape.

Reply:
> You should still include the "Indexing starts at 1". Otherwise it will result into an error.
he meant the permute starts from index 1 of the tensor as index 0 refers to the batch which need not be included.

Reply:
Please wrap the long line here.